### DISCUSSION TITLE
TCP 와 UDP

### DISCUSSION THREAD

**Discussion by WooGie911** [wanted-9th-3team/frontend-study#56] [프론트 기술 질문 공간] [answered]
TCP 와 UDP 를 비교해서 설명해주세요.

**Comment by WooGie911** [comment, marked as answer]
TCP는 신뢰성 있는 데이터 전송을 지원하는 연결 지향형 프로토콜입니다. 
일반적으로 TCP와 IP가 함께 사용되는데, IP가 데이터의 전송을 처리한다면 TCP는 패킷 추적 및 관리를 하게 됩니다. 

UDP는 비연결형 프로토콜로써, 인터넷상에서 서로 정보를 주고받을 때 정보를 보낸다는 신호나 받는다는 신호 절차를 거치지 않고 보내는 쪽에서 일방적으로 데이터를 전달하는 통신 프로토콜입니다. 
TCP와는 다르게 연결 설정이 없으며, 혼잡 제어를 하지 않기 때문에 TCP보다 전송 속도가 빠릅니다. 
그러나 데이터 전송에 대한 보장을 하지 않기 때문에 패킷 손실이 발생할 수 있습니다.

**Comment by michoball** [comment]
TCP는 IP와 함께 TCP/IP라고 불립니다. TCP는 네트워킹에 쓰이는 프로토콜입니다. TCP는 네트워크 상의 두 장치간에 신뢰성있는 연결을 지향합니다.  두 장치를 연결하여 전송 확인과 응답으로 데이터를 신뢰성있게 전달할 때 쓰입니다. 예를 들면 이미지와 같은 파일을 네트워크 전송을 할 때 파일을 분해해서 보내게 되는데 이를 재조립하고 빠진 부분이 있다면 재요청을 보냅니다.  

UDP는 비연결형으로 누가 정보를 주고 받는지에 대한 내용을 가지지 않고 아무나 보내오면 그냥 다 받는 식입니다. 데이터가 신뢰성을 가지기 어렵기 때문에 중요한 데이터를 주고받는데 쓰이지 않고 실시간 스트리밍이나 DNS에서 데이터를 전송하는데 사용됩니다.

**Comment by javascriptbiggosoo** [comment]
![image](https://github.com/wanted-9th-3team/frontend-study/assets/28939278/f284a244-a931-4f09-a177-fe1c423af424)
네트워크 개념상 HTTP 프로토콜은 TCP 위의 계층이며, 동시에 TCP 는 IP의 상위 계층입니다.
HTTP가 출시되기 이전부터 TCP는 메시지 데이터를 전송하는데에 대중적이고 신뢰성을 갖춘 전송 프로토콜로서 역할을 해냈기 때문에 HTTP에서 이를 채용했고 네트워크 통신의 핵심적인 세부사항을 일임하였습니다.


UDP는 피시방에서 스타크래프트 친구들이랑 같이할 때 써봤습니다.
